services/explorer/src/pages/Holders/Holders.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import {
+  Box,
+  TablePagination,
+  TableContainer,
+  Table,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody,
+  EmptyTable,
+} from '@taraxa_project/taraxa-ui';
+import { PageTitle } from '../../components';
+import { useHoldersEffects } from './Holders.effects';
+
+const HoldersPage = (): JSX.Element => {
+  const {
+    title,
+    description,
+    cols,
+    rows,
+    rowsPerPage,
+    page,
+    handleChangePage,
+    handleChangeRowsPerPage,
+    pagination,
+  } = useHoldersEffects();
+  return (
+    <>
+      <PageTitle title={title} subtitle={description} />
+      <Box sx={{ display: 'flex', flexDirection: 'column', gap: '2rem' }}>
+        <TablePagination
+          rowsPerPageOptions={[25, 50, 75, 100]}
+          component='div'
+          count={pagination.total}
+          rowsPerPage={rowsPerPage}
+          page={page}
+          onPageChange={(
+            event: React.MouseEvent<HTMLButtonElement> | null,
+            page: number
+          ) => {
+            handleChangePage(page);
+          }}
+          onRowsPerPageChange={(event: React.ChangeEvent<HTMLInputElement>) =>
+            handleChangeRowsPerPage(parseInt(event.target.value, 10))
+          }
+        />
+        <TableContainer sx={{ marginBottom: '2rem' }}>
+          <Table style={{ tableLayout: 'auto', marginBottom: '2rem' }}>
+            <TableHead>
+              <TableRow tabIndex={-1} key='index'>
+                {cols.map((column, index) => (
+                  <TableCell key={`${index}-${index}-head`}>
+                    {column.name}
+                  </TableCell>
+                ))}
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {rows && rows.length > 0 ? (
+                rows.map((row, index) => (
+                  <TableRow tabIndex={-1} data-key={index} key={index}>
```

Comment:
We can use the address here instead of the index

Reply:
Address is a JSX.Element here but we can use a combination of `${index}-${value}` that might be better

Reply:
That won't work either. Maybe we can just convert the address to a string and add the element here?

Reply:
Apparently `row.address.props.hash` works. 👍🏼